examples/server/server.cpp

```diff
@@ -2115,12 +2115,17 @@ struct server_context {
                         }
 
                         if (slot.cmpl_type == SERVER_TASK_CMPL_TYPE_EMBEDDING || slot.cmpl_type == SERVER_TASK_CMPL_TYPE_RERANK) {
-                            // this prompt is too large to process - discard it
                             if (slot.n_prompt_tokens > n_ubatch) {
                                 slot.release();
                                 send_error(slot, "input is too large to process. increase the physical batch size", ERROR_TYPE_SERVER);
                                 continue;
                             }
+
+                            if (slot.n_prompt_tokens > slot.n_ctx) {
+                                slot.release();
+                                send_error(slot, "input is larger than the max context size. skipping", ERROR_TYPE_SERVER);
```

Comment:
I'd suggest changing some wordings to match the message when number of tokens is larger than `n_ubatch`

```suggestion
                                send_error(slot, "input is too large to process. increase the context size", ERROR_TYPE_SERVER);
```

Reply:
Suggesting to increase the context size can lead to the same issue as the user experienced in #9978 due to setting a context size larger than what the model supports. So I'll leave the version without the suggestion.